src/@next/Select/components/Activator/ActivatorSelect.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useRef } from 'react';
+import { Icon } from '../../../Icon';
+
+import { Typography } from '../../../Typography';
+import { Blue, Neutral } from '../../../utilities/colors';
+import {
+  ActivatorSelectContextProps,
+  useSelectActivator,
+} from './ActivatorContext';
+import { StyledSelect, StyledTag, TagsContainer } from './ActivatorStyle';
+
+export interface ActivatorSelectProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  value?: string;
+  placeholder?: string;
+  onRemoveTag?({ option }: { option: string }): void;
+  width?: number;
+}
+export const ActivatorSelect = ({
+  placeholder,
+  value,
+  onClick,
+  onRemoveTag,
+  width,
+  ...props
+}: ActivatorSelectProps) => {
+  const activatorRef = useRef(null);
+  const hasValue = !!value;
+  const activatorContext = useSelectActivator() as ActivatorSelectContextProps;
+  const {
+    allowMultiple,
+    disabled,
+    onSelectClick,
+    hasError,
+    setWidth,
+    selectedValues,
+  } = activatorContext;
+
+  useEffect(() => {
+    setWidth(activatorRef.current.getBoundingClientRect().width);
+  }, [activatorRef, setWidth]);
```

Comment:
not sure what you mean, they're both used in line 40?

Reply:
Oh right my bad, didn't see the setWidth. I guess it's strange because usually ref.current is used to escape the dependencies, does the exhaustive depth give you a warning if you remove activatorRef?